mypy/fastparse.py

```diff
@@ -112,6 +116,23 @@ def parse_type_comment(type_comment: str, line: int, errors: Optional[Errors]) -
         return TypeConverter(errors, line=line).visit(typ.body)
 
 
+def parse_docstring(hook: DocstringParserHook, docstring: str, arg_names: List[str],
+                    line: int, errors: Errors) -> Optional[Tuple[List[Type], Type]]:
+    """Parse a docstring and return type representations.
+
+    Returns a 2-tuple: (list of arguments Types, and return Type).
+    """
+    type_map = hook(docstring, line, errors)
+    if type_map:
+        arg_types = [type_map.pop(name, AnyType()) for name in arg_names]
+        return_type = type_map.pop('return', AnyType())
+        if type_map:
+            errors.report(line, 0,
+                          TYPE_COMMENT_DOCSTRING_ERROR.format(list(type_map), arg_names))
```

Comment:
> Maybe there also ought to be an error for arguments in the arg list but not in the docstring (given that there's a type map at all).

I think it's important that docstring annotations can be sparse, as with multi-line python 2 function annotations.